Sources/SwiftParser/Expressions.swift

```diff
@@ -1746,7 +1746,9 @@ extension Parser {
     var effectSpecifiers: RawTypeEffectSpecifiersSyntax?
     var returnClause: RawReturnClauseSyntax? = nil
     if !self.at(.keyword(.in)) {
-      if self.at(.leftParen) {
+      // If the next token is ':', then it looks like the code contained a non-shorthand closure parameter with a type annotation.
+      // These need to be wrapped in parentheses.
+      if self.at(.leftParen) || self.peek(isAt: .colon) {
```

Comment:
I added this test and it fails because we are peeking here. 
When at this place current token is `comparisonPredicate`.

```swift
    assertParse(
      """
      withInvalidOrderings { (comparisonPredicate: @escaping (Int, Int) -> Bool) in
      }
      """
    )
```

So the peek seems not the correct solutions here. 
Maybe we should do something like `self.peek(isAt: .colon) && previousToken != .leftParen`

What do you think @ahoppen

Reply:
@ahoppen 😬

Reply:
Sorry, it’s been a busy week and I missed the comment. Thanks for the ping.

The issue happens because in `canPareClosureSignature`, we now stop at the arrow. Since we start doing `skipSingle` at `comparisonPredicate`. If you make the following change, then we’re calling `skipSingle` on the opening `(`, which consumes all the contents until the matching `)` before the `in`, including the `->`, which fixes the issue.

```diff
- if lookahead.consume(if: .leftParen) != nil {  // Consume the ')'.
+ if lookahead.at(.leftParen) {  // Consume the '('.
```

Reply:
Oh okay nice! 
Thanks a lot. 

And no worries! Take the time to reply when you can 😁